server/src/main/java/io/deephaven/server/barrage/BarrageMessageProducer.java

```diff
@@ -558,6 +583,12 @@ public boolean updateSubscription(
             if (sub.pendingViewport != null) {
                 sub.pendingViewport.close();
             }
+            if (sub.isFullSubscription() != (newViewport == null)) {
```

Comment:
1. This seems unnecessarily permissive. Once you've become fullSubscription, you should not be able to hit this method, I think.
2. Does this make sense to do after you've already checked the null-ness of pending viewport? I guess it doesn't matter, since we're closing it but not nulling it out.